pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -749,13 +750,35 @@ func (s *Service) DiscoverLaunchTemplateAMI(scope scope.LaunchTemplateScope) (*s
 		imageLookupBaseOS = scope.GetEC2Scope().ImageLookupBaseOS()
 	}
 
+	instanceType := lt.InstanceType
+	// If instance type is not specified on a launch template, we can safely assume the instance type will be a `t3.medium`.
+	// As specified in the AWS docs https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html.
+	if instanceType == "" {
+		instanceType = "t3.medium"
+	}
```

Comment:
I found the issue with the AMI population @Ankitasw, it looks like when running the tests with a launch template, the `instanceType` field is left empty, this is because AWS automatically populates this with `t3.medium` on their end when using a launch template.

I've added a check here to just assume the instanceType is a `t3.medium` in this case.